src/textual/widgets/_button.py

```diff
@@ -100,3 +205,93 @@ async def on_click(self, event: events.Click) -> None:
     async def on_key(self, event: events.Key) -> None:
         if event.key == "enter" and not self.disabled:
             await self.emit(Button.Pressed(self))
+
+    @staticmethod
+    def success(
```

Comment:
I've added both the `variant` constructor arg and these utility constructors. I'm not sure which is best. The utility constructors feel like they may be a bit of a maintenance burden with keeping the signatures in sync (unless we do some `getattr` craziness).

Happy to go with either or both, or none if anyone has other thoughts.

Reply:
I'm happy with both. It is a maintenance burden, but we suffer so Textual devs can save a second or two.